libsledge/src/table_instructions.c

```diff
@@ -0,0 +1,49 @@
+#include "sledge_abi.h"
+
+#define unlikely(X)         __builtin_expect(!!(X), 0)
+#define INDIRECT_TABLE_SIZE (1 << 10)
+#define INLINE              __attribute__((always_inline))
+#define likely(X)           __builtin_expect(!!(X), 1)
+#define unlikely(X)         __builtin_expect(!!(X), 0)
+
+/* This is private and NOT in the sledge_abi.h header because the runtime uses an overlay struct that extends this
+ * symbol with private members */
+extern thread_local struct sledge_abi__wasm_module_instance sledge_abi__current_wasm_module_instance;
+
+static INLINE void *
+wasm_table_get(struct sledge_abi__wasm_table *wasm_table, uint32_t idx, uint32_t type_id)
+{
+	assert(wasm_table != NULL);
+	assert(idx < wasm_table->capacity);
+
+	struct sledge_abi__wasm_table_entry f = wasm_table->buffer[idx];
+
+	// assert(f.type_id != type_id);
```

Comment:
What's this?

Reply:
This is commented out because Wasmception compiled serverless functions trigger type errors. I added a note into the source and expect this to be reenabled in the follow-up WASI PR.